unlock_indicator.c

```diff
@@ -229,6 +234,30 @@ xcb_pixmap_t draw_image(uint32_t *resolution) {
             cairo_close_path(ctx);
         }
 
+        if (pam_state == STATE_PAM_WRONG && (capslock_active || numlock_active)) {
+            cairo_text_extents_t extents;
+            double x, y;
+            char lock_text[32] = "";
```

Comment:
It seems like a nitpick at first, but considering i3lock’s security-sensitive nature, I’d like to avoid statically sized buffers and `strcat()`, so please change this to something like:

```c
char *lock_text = NULL;
if (asprintf(&lock_text, "%s%s%s locked",
    (capslock_active ? "Caps" : ""),
    (capslock_active && numlock_active ? ", " : ""),
    (numlock_active ? "Num" : "")) != -1) {
  // cairo code…
  free(lock_text);
}
```